### ISSUE TITLE
Wrong dataset URL throws 500 instead of 404

### ISSUE THREAD

**Issue by arunapa** [clowder-framework/clowder2#380]
hm actually i can't trigger this 404. whenever I gave a wrong URL, i get 500 error. E.g.
```
role = AuthorizationDB.from_mongo(authorization).role
AttributeError: 'NoneType' object has no attribute 'role'
```

_Originally posted by @longshuicy in https://github.com/clowder-framework/clowder2/issues/371#issuecomment-1454232211_

**Comment by arunapa** [comment]
```
wait call(**sub_values)
  File "/Users/cwang138/Documents/Clowder/clowder2/backend/./app/deps/authorization_deps.py", line 58, in __call__
    "dataset_id": ObjectId(dataset_id),
  File "/Users/cwang138/.local/share/virtualenvs/backend-pkZ6g-7g/lib/python3.9/site-packages/bson/objectid.py", line 103, in __init__
    self.__validate(oid)
  File "/Users/cwang138/.local/share/virtualenvs/backend-pkZ6g-7g/lib/python3.9/site-packages/bson/objectid.py", line 203, in __validate
    _raise_invalid_id(oid)
  File "/Users/cwang138/.local/share/virtualenvs/backend-pkZ6g-7g/lib/python3.9/site-packages/bson/objectid.py", line 35, in _raise_invalid_id
    raise InvalidId(
bson.errors.InvalidId: '64027a0c47ba023222d440' is not a valid ObjectId, it must be a 12-byte input or a 24-character hex string
INFO:     127.0.0.1:61918 - "OPTIONS /api/v2/datasets/64027a0c47ba023222d44093/files?limit=10 HTTP/1.1" 200 OK
```

**Comment by arunapa** [comment]
```
INFO:     127.0.0.1:50254 - "GET /api/v2/datasets/64011ca928b47b7ff62a9c6b HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 407, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/fastapi/applications.py", line 270, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/starlette/middleware/cors.py", line 92, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/starlette/middleware/cors.py", line 147, in simple_response
    await self.app(scope, receive, send)
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/starlette/routing.py", line 706, in __call__
    await route.handle(scope, receive, send)
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/fastapi/routing.py", line 227, in app
    solved_result = await solve_dependencies(
  File "/Users/aruna/.local/share/virtualenvs/backend-0B_NbkUZ/lib/python3.9/site-packages/fastapi/dependencies/utils.py", line 543, in solve_dependencies
    solved = await call(**sub_values)
  File "/Users/aruna/ws/ncsa/clowder2/backend/./app/deps/authorization_deps.py", line 63, in __call__
    role = AuthorizationDB.from_mongo(authorization).role
AttributeError: 'NoneType' object has no attribute 'role'
```

**Comment by longshuicy** [comment]
> 

I think i know why this happens. We need to cast dataset_id from string to objectId. Might be addressed in this PR https://github.com/clowder-framework/clowder2/pull/354

**Comment by longshuicy** [comment]
@arunapa could you test with the latest merged in PR #354 and see if this got addressed

**Comment by arunapa** [comment]
@longshuicy Verified that this issue is no longer reproable. I provided an incorrect dataset URL, and it redirected me to the 404 page instead.
